java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/ClientCredentialsTokenRequestBuilder.java

```diff
@@ -0,0 +1,62 @@
+package co.worklytics.psoxy.gateway.impl.oauth;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import co.worklytics.psoxy.gateway.RequiresConfiguration;
+import com.google.api.client.http.HttpContent;
+import com.google.api.client.http.UrlEncodedContent;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * implementation of an Access Token Request (per RFC6749 4.4.2) authenticated by an assertion (RFC 7521) based
+ * on client id + client secret
+ *
+ * see
+ *   - https://datatracker.ietf.org/doc/html/rfc7521
+ *   - https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2
+ */
+@NoArgsConstructor(onConstructor_ = @Inject)
+public class ClientCredentialsTokenRequestBuilder
```

Comment:
Added this and renamed `ClientCredentialsGrantTokenRequestBuilder` to `ClientCredentialsWithJWTAssertionGrantTokenRequestBuilder`; the idea is to support and differenciate two scenarios of same client credentials grant type: with client id + client secret to obtain a token and the one used for building a JWT assertion based on a public/private key.

ClientCredentialsWithJWTAssertionGrantTokenRequestBuilder is not currently used; it was the one used by MSFT connectors but since workload federation no one uses currently in current specs, but I think is interesting to keep it in anycase.